instrumentation/ratpack-1.4/library/src/main/java/io/opentelemetry/instrumentation/ratpack/RatpackProductionErrorHandler.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+// Includes work from:
+/*
+ * Copyright 2014 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.ratpack;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ratpack.error.ClientErrorHandler;
+import ratpack.error.ServerErrorHandler;
+import ratpack.handling.Context;
+
+// Copied from
+// https://github.com/ratpack/ratpack/blob/master/ratpack-core/src/main/java/ratpack/core/error/internal/DefaultProductionErrorHandler.java
+// since it is internal and has had breaking changes.
+final class RatpackProductionErrorHandler implements ClientErrorHandler, ServerErrorHandler {
+
+  static final RatpackProductionErrorHandler INSTANCE = new RatpackProductionErrorHandler();
+
+  private static final Logger logger = LoggerFactory.getLogger(RatpackProductionErrorHandler.class);
```

Comment:
This is using "our" slf4j, not ratpack's slf4j. Is that okay?

Reply:
Hmm good point. In practice I don't think it's possible to use this class as the context should always have a error handler I think. Should I verify that and throw an exception outside of the loop that finds `delegate` instead of this fallback? Maybe slightly dangerous if there is some unexpected ratpack behavior though.

Reply:
Hmm, maybe just logging something would be enough? And a bit safer

Reply:
Added a WarnOnce